static/flow/views/data-set-view.js

```diff
@@ -199,7 +199,15 @@ var DataSetView = function(options) {
             displayedName = base.m_dataSet.metadata.displayedName;
         }
         info.append($('<div>').text("DataSet Name: " + displayedName));
-        info.append($('<div>').text("Program Name: " + base.m_dataSet.metadata.program.name));
+        if(base.m_dataSet.metadata.program.displayedName == null){
+            //old style dataset where name/displayed name are the same
+            //and displayedName value is null or undefined
+            info.append($('<div>').text("Program Name: " + base.m_dataSet.metadata.program.name)); 
+        }
+        else{
+            //modern style dataset where name/displayed name are unique
+            info.append($('<div>').text("Program Name: " + base.m_dataSet.metadata.program.displayedName));            
```

Comment:
Super small, but I'd probably factor out the `displayName` into a `const` to try and avoid duplicating the code in these branches.

Reply:
Do you mean the other git branch or the branch of the if-else?  I don't quite follow.

Reply:
The if-else branch, sorry for the confusion!

Reply:
Oh, I think you mean the internal name.  Sounds good.

Reply:
I'll post a code example to clarify.  I see what you're saying.

Reply:
Something like this:
`		var programDisplayedName;
        if(base.m_dataSet.metadata.program.displayedName == null){
            //old style dataset where name/displayed name are the same
            //and displayedName value is null or undefined
			programDisplayedName = base.m_dataSet.metadata.program.name;
            info.append($('<div>').text("Program Name: " + base.m_dataSet.metadata.program.name)); 
        }
        else{
            //modern style dataset where name/displayed name are unique
			programDisplayedName = base.m_dataSet.metadata.program.displayedName;
        }
        info.append($('<div>').text("Program Name: " + programDisplayedName));      `

Reply:
That didn't display quite right, but I think you get the picture.

Reply:
Anyway, I pushed a change that reduces the redundant code by storing the name in a var.